src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -544,6 +544,60 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
             })
     }
 
+    /// Resolve a type alias for the purposes of intra-doc links.
+    fn resolve_type_alias(&self, ty_id: DefId) -> Option<Res> {
+        use PrimitiveType::*;
+        let ty = self.cx.tcx.type_of(ty_id);
+        Some(match ty.kind() {
+            ty::Bool => Res::Primitive(Bool),
+            ty::Char => Res::Primitive(Char),
+            ty::Int(ity) => Res::Primitive(match ity {
+                ty::IntTy::Isize => Isize,
+                ty::IntTy::I8 => I8,
+                ty::IntTy::I16 => I16,
+                ty::IntTy::I32 => I32,
+                ty::IntTy::I64 => I64,
+                ty::IntTy::I128 => I128,
+            }),
```

Comment:
You've written this match before, you can reuse it: https://github.com/rust-lang/rust/blob/2336406b38db20d1bad30d32914e73af1dd62742/src/librustdoc/clean/types.rs#L1782
```suggestion
            ty::Int(ity) => Res::Primitive(ity.into()),
```

Reply:
It's actually not the same type anymore (`ast::IntTy` vs `ty::IntTy`)

Reply:
Oops, nevernmind, there's also https://github.com/rust-lang/rust/blob/2336406b38db20d1bad30d32914e73af1dd62742/src/librustdoc/clean/types.rs#L1817